cannot install or build tectonic

when `cargo install tectnoic` or `cargo build`, get error as following:

Internal error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-m64" "-I" "." "-I" "/home/less/IdeaProjects/tectonic/target/debug/build/tectonic-ab19c26fd83db5db/out" "-I" "/usr/include/freetype2" "-I" "/usr/include/harfbuzz" "-I" "/usr/include/x86_64-linux-gnu" "-I" "/usr/include/freetype2" "-I" "/usr/include/libpng12" "-Wall" "-Wextra" "-Wall" "-Wcast-qual" "-Wdate-time" "-Wendif-labels" "-Wextra" "-Wformat=2" "-Winit-self" "-Wlogical-op" "-Wmissing-declarations" "-Wmissing-include-dirs" "-Wmissing-prototypes" "-Wnested-externs" "-Wold-style-definition" "-Wpointer-arith" "-Wredundant-decls" "-Wstrict-prototypes" "-Wsuggest-attribute=format" "-Wswitch-bool" "-Wundef" "-Wwrite-strings" "-Wno-unused-parameter" "-Wno-implicit-fallthrough" "-Wno-sign-compare" "-std=gnu11" "-DHAVE_ZLIB=1" "-DHAVE_ZLIB_COMPRESS2=1" "-DZLIB_CONST=1" "-o" "/home/less/IdeaProjects/tectonic/target/debug/build/tectonic-ab19c26fd83db5db/out/tectonic/core-bridge.o" "-c" "tectonic/core-bridge.c" with args "cc" did not execute successfully (status code exit code: 1).

Yes, the error messge is as following:

cargo:warning=In file included from tectonic/xetexd.h:11:0,
cargo:warning=                 from tectonic/core-bridge.c:15:
cargo:warning=tectonic/XeTeXOTMath.h:43:5: error: unknown type name ‘hb_ot_math_glyph_part_t’
cargo:warning=     hb_ot_math_glyph_part_t *parts;
cargo:warning=     ^
cargo:warning=cc1: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
exit code: 1

--- stderr
thread 'main' panicked at '

Internal error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-m64" "-I" "." "-I" "/home/less/IdeaProjects/tectonic/target/debug/build/tectonic-ab19c26fd83db5db/out" "-I" "/usr/include/freetype2" "-I" "/usr/include/harfbuzz" "-I" "/usr/include/x86_64-linux-gnu" "-I" "/usr/include/freetype2" "-I" "/usr/include/libpng12" "-Wall" "-Wextra" "-Wall" "-Wcast-qual" "-Wdate-time" "-Wendif-labels" "-Wextra" "-Wformat=2" "-Winit-self" "-Wlogical-op" "-Wmissing-declarations" "-Wmissing-include-dirs" "-Wmissing-prototypes" "-Wnested-externs" "-Wold-style-definition" "-Wpointer-arith" "-Wredundant-decls" "-Wstrict-prototypes" "-Wsuggest-attribute=format" "-Wswitch-bool" "-Wundef" "-Wwrite-strings" "-Wno-unused-parameter" "-Wno-implicit-fallthrough" "-Wno-sign-compare" "-std=gnu11" "-DHAVE_ZLIB=1" "-DHAVE_ZLIB_COMPRESS2=1" "-DZLIB_CONST=1" "-o" "/home/less/IdeaProjects/tectonic/target/debug/build/tectonic-ab19c26fd83db5db/out/tectonic/core-bridge.o" "-c" "tectonic/core-bridge.c" with args "cc" did not execute successfully (status code exit code: 1).

--- a/.circleci/outer-build.sh
+++ b/.circleci/outer-build.sh
@@ -5,7 +5,7 @@
 set -e -x
 
 buildroot=$HOME
-tarball='https://bintray.com/pkgw/tectonic/download_file?file_path=tectonic-buildenv-ppc-20180128.tar.gz'
+tarball='https://dataverse.harvard.edu/api/access/datafile/:persistentId?persistentId=doi:10.7910/DVN/VM4ZZ3/NT9HFL'
 
 # Validate that our little scheme is going to work. Annoyingly
 # $CIRCLE_WORKING_DIRECTORY (passed in as $1 since we're sudo) is defined with

--- a/build.rs
+++ b/build.rs
@@ -15,7 +15,7 @@ use std::path::PathBuf;
 // MacOS platform specifics:
 
 #[cfg(target_os = "macos")]
-const LIBS: &'static str = "harfbuzz harfbuzz-icu icu-uc freetype2 graphite2 libpng zlib";
+const LIBS: &'static str = "harfbuzz >= 1.4 harfbuzz-icu icu-uc freetype2 graphite2 libpng zlib";
 
 #[cfg(target_os = "macos")]
 fn c_platform_specifics(cfg: &mut cc::Build) {
@@ -40,7 +40,7 @@ fn cpp_platform_specifics(cfg: &mut cc::Build) {
 // Not-MacOS:
 
 #[cfg(not(target_os = "macos"))]
-const LIBS: &'static str = "fontconfig harfbuzz harfbuzz-icu icu-uc freetype2 graphite2 libpng zlib";
+const LIBS: &'static str = "fontconfig harfbuzz >= 1.4 harfbuzz-icu icu-uc freetype2 graphite2 libpng zlib";
 
 #[cfg(not(target_os = "macos"))]
 fn c_platform_specifics(_: &mut cc::Build) {

